cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/batch/MockSource.java

```diff
@@ -175,6 +214,52 @@ private static PluginClass getPluginClass() {
     Map<String, PluginPropertyField> properties = new HashMap<>();
     properties.put("tableName", new PluginPropertyField("tableName", "", "string", true, false));
     properties.put("schema", new PluginPropertyField("schema", "", "string", false, false));
+    properties.put("metadataOperations", new PluginPropertyField("metadataOperations", "", "string", false, false));
     return new PluginClass(BatchSource.PLUGIN_TYPE, "Mock", "", MockSource.class.getName(), "config", properties);
   }
+
+  /**
+   * Processes metadata operations
+   */
+  private void processsMetadata(BatchSourceContext context) throws MetadataException {
+    MetadataEntity metadataEntity = MetadataEntity.ofDataset(context.getNamespace(), config.tableName);
+    Map<MetadataScope, Metadata> curMetadata =
+      context.getMetadata(metadataEntity);
+    Set<MetadataOperation> operations = GSON.fromJson(config.metadataOperations, SET_METADATA_OPERATION_TYPE);
+    // must be to fetch metadata and there should be system metadata
+    if (curMetadata.get(MetadataScope.SYSTEM).getProperties().isEmpty() ||
+      curMetadata.get(MetadataScope.SYSTEM).getProperties().isEmpty()) {
+      throw new RuntimeException(String.format("System properties or tags for '%s' is empty. " +
```

Comment:
Shouldn't it be `IllegalArgumentException`?

Reply:
Done.